public/app/features/dimensions/resource.ts

```diff
@@ -40,17 +40,18 @@ export function getResourceDimension(
   }
 
   if (mode === ResourceDimensionMode.Mapping) {
-    const mapper = (v: any) => getPublicOrAbsoluteUrl(`${v}`);
+    const mapper = (v: string) => getPublicOrAbsoluteUrl(`${v}`);
     return {
       field,
       get: (i) => mapper(field.values.get(i)),
       value: () => mapper(getLastNotNullFieldValue(field)),
     };
   }
 
-  const getIcon = (value: any): string => {
-    const disp = field.display!;
-    return getPublicOrAbsoluteUrl(disp(value).icon ?? '');
+  // mode === ResourceDimensionMode.Field case
+  const getIcon = (value: string): string => {
+    const display = field.display!;
+    return getPublicOrAbsoluteUrl(display(value).icon ?? value ?? '');
```

Comment:
The additional fallback here ` ?? value ` is what fixed the issue. This solution does have an issue though when the path is invalid / url doesn't allow CORS (displays empty instead falling back to `?` icon)